lib/repl.js

```diff
@@ -282,17 +282,24 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string')) {
+      out = e;
```

Comment:
Symbols and other bad inputs are taken care of here if I move to `out = util.inspect(e);`. However, it causes `test-repl-sigint.js` and `test-repl-sigint-nested-eval.js` to fail. @addaleax any idea why that would be?